pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -229,6 +329,15 @@ func (ic *ifConfigurator) removeHNSEndpoint(endpoint *hcsshim.HNSEndpoint, conta
 	deleteCh := make(chan error)
 	// Remove HNSEndpoint.
 	go func() {
+		hcnEndpoint, _ := hcn.GetEndpointByID(endpoint.Id)
+		if hcnEndpoint != nil && hcnEndpoint.HostComputeNamespace != "" {
+			err := hcn.RemoveNamespaceEndpoint(hcnEndpoint.HostComputeNamespace, hcnEndpoint.Id)
+			if err != nil {
+				klog.Errorf("Failed to remove HostComputeEndpoint %s from HostComputeNameSpace %s: %v", hcnEndpoint.Name, hcnEndpoint.HostComputeNamespace, err)
```

Comment:
looks like the error will be logged a second time below, maybe use:
```go
deleteCh <- fmt.Errorf("failed to remove HostComputeEndpoint %s from HostComputeNamespace %s: %v", hcnEndpoint.Name, hcnEndpoint.HostComputeNamespace, err)
```

Reply:
```
	case err := <-deleteCh:
		if err != nil {
			if !strings.Contains(err.Error(), notFoundHNSEndpoint) {
				klog.Errorf("Failed to delete container interface %s: %v", containerID, err)
				return err
			}
		}
	case <-time.After(1 * time.Second):
		return fmt.Errorf("timeout when deleting HNSEndpoint %s", epName)
	}
```

The `select` will exist if the excution time exceed 1 second.  After the change. If the error happens after 1 second, the error won't get printted. what do you think

Reply:
I see, thanks for pointing it out.

Then maybe let's always log the error in the goroutine (also for the call to `endpoint.Delete` below), assuming it is an actual error (and not "not found").

Then, also assuming it is an actual error, we can write it to the `deleteCh` channel. To avoid double logging, we can 
change the select case for `deleteCh` to:
```go
	case err := <-deleteCh:
		if err != nil {
			return err
		}
```

What do you think?